src/RequestInsurance/Controllers/RequestInsuranceEditController.php

```diff
@@ -0,0 +1,171 @@
+        }
+        $requestInsuranceEdit->delete();
+
+        return redirect()->back();
+    }
+
+    /**
+     * @param Request $request
+     * @param RequestInsuranceEdit $requestInsuranceEdit
+     * @param IdentityProvider $identityProvider
+     *
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function update(Request $request, RequestInsuranceEdit $requestInsuranceEdit, IdentityProvider $identityProvider)
+    {
+        // Only allow updates if it has not been applied and the updating user is the edit author
+        if ($requestInsuranceEdit->applied_at != null || $identityProvider->getUser($request) != $requestInsuranceEdit->admin_user) {
+            // Both these cases should not be possible from the view, so we don't send any error message
+            return redirect()->back();
+        }
+
+        DB::transaction(function () use ($request, $requestInsuranceEdit) {
+            // Remove all approvals
+            $requestInsuranceEdit->approvals()->delete();
+
+            // Update the edit
+            $requestInsuranceEdit->update([
+                'new_priority'         => $request->post('new_priority', $requestInsuranceEdit->new_priority),
+                'new_url'              => $request->post('new_url', $requestInsuranceEdit->new_url),
+                'new_method'           => $request->post('new_method', $requestInsuranceEdit->new_method),
+                'new_headers'          => $request->post('new_headers', ''),
+                'new_payload'          => $request->post('new_payload', ''),
+                'new_encrypted_fields' => $request->post('new_encrypted_fields', ''),
+            ]);
+        });
+
+        return redirect()->back();
+    }
+
+    /**
+     * @param RequestInsuranceEdit $requestInsuranceEdit
+     *
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function apply(RequestInsuranceEdit $requestInsuranceEdit)
+    {
+        // If already applied, do nothing
+        if ($requestInsuranceEdit->applied_at != null) {
+            return redirect()->back();
+        }
+
+        $errors = [];
+
+        if ($requestInsuranceEdit->approvals()->count() < $requestInsuranceEdit->required_number_of_approvals) {
+            $errors['requestInsuranceEdit'] = $requestInsuranceEdit;
+            $errors['requestErrors'] = ['approval' => 'Not enough approvals to apply'];
+        }
+
+        if ( ! $this->is_valid_header_format($requestInsuranceEdit->new_headers)) {
+            $errors['requestInsuranceEdit'] = $requestInsuranceEdit;
+            $errors['requestErrors'] = ['header' => 'Invalid header format'];
+        }
+
+        // If any errors redirect back with all errors
+        if ( ! empty($errors)) {
+            return redirect()->back()->with($errors);
+        }
+
+        DB::transaction(function () use ($requestInsuranceEdit) {
+            $requestInsuranceEdit->update(['applied_at' => Carbon::now()]);
```

Comment:
Jeg har lavet et kort der tvinger RI til at bruge UTC :smile: Så alle steder du bruger Carbon::now() der skal du lige smide UTC ind som TZ

```suggestion
            $requestInsuranceEdit->update(['applied_at' => Carbon::now('UTC')]);
```